From e5baabf57b1f6bcaf60068b3fd24af649e4e91c6 Mon Sep 17 00:00:00 2001
From: Joseph Myers <joseph@codesourcery.com>
Date: Wed, 19 Aug 2020 22:46:41 +0000
Subject: [PATCH 032/267] Add C2x BOOL_MAX and BOOL_WIDTH to limits.h.

C2x adds BOOL_MAX and BOOL_WIDTH macros to <limits.h>.  This patch
adds them to glibc's <limits.h> for the case when they aren't defined
by GCC's <limits.h>.

Tested for x86_64.
---
 include/limits.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/include/limits.h b/include/limits.h
index 8195da78a4..29e9b943b9 100644
--- a/include/limits.h
+++ b/include/limits.h
@@ -178,6 +178,18 @@
 # endif
 #endif /* Use IEC_60559_BFP_EXT.  */
 
+/* The macros for _Bool are not defined by GCC's <limits.h> before GCC
+   11, or if _GNU_SOURCE is defined rather than enabling C2x support
+   with -std.  */
+#if __GLIBC_USE (ISOC2X)
+# ifndef BOOL_MAX
+#  define BOOL_MAX 1
+# endif
+# ifndef BOOL_WIDTH
+#  define BOOL_WIDTH 1
+# endif
+#endif
+
 #ifdef	__USE_POSIX
 /* POSIX adds things to <limits.h>.  */
 # include <bits/posix1_lim.h>
-- 
2.29.0

